VimeoNetworking/VimeoNetworking/ResponseCache.swift

```diff
@@ -35,22 +48,18 @@ final internal class ResponseCache
     {
         let key = request.cacheKey
         
-        if let responseDictionary = self.memoryCache.responseDictionaryForKey(key)
-        {
-            completion(result: .Success(result: responseDictionary))
-        }
-        else
-        {
-            self.diskCache.responseDictionaryForKey(key) { responseDictionary in
-                
-                if let responseDictionary = responseDictionary
-                {
+        self.memoryCache.responseDictionary(forKey: key) { (responseDictionary) in
```

Comment:
I really like the idea of having `Cache`s which have a shared, structured contract. This is just syntax which has no functional impact. I can remove the protocol if you feel strongly about it though.